### PR TITLE
Feature/auth

### PR DESCRIPTION
**Pull request by xal9wiii4ik**
Authorization/registration

as user i want to create my account for creating new videos or something else in future and i want to be sure that my username or email using only by myself, and i want to get or edit my account data
as frontend developer i want to see what endpoints we have and what data i should put in

added the ability to create a new account and auxiliary functions for frontend developer (swagger);
updated tests, docker(poetry instead of pip); added simple register new account validations, added authorization(ability to getting access tokens and refresh)

### CODE REVIEW

File: app/api/auth/services.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+
+def create_new_user(data: tp.Dict[str, tp.Any]) -> None:
+    """
+    Create new user and update returned user data
+    Args:
+        data: new user data
+    """
+
+    del data['repeat_password']
```
Comment (xal9wiii4ik) [inline app/api/auth/services.py:33]: deleting data, since this dict is copied from the request.json by serializer

File: .gitignore [hunk 2]
```diff
@@ -215,4 +215,6 @@ ENV/
 media_volume
 static_volume
 
+poetry.lock
```
Comment (artsiom-streetshares) [inline .gitignore]: this should never be in gitignore. this is the whole point of poetry

File: app/api/auth/serializers.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+from base import settings
+
+
+class RegisterUserSerializer(serializers.Serializer):
+    """
+    Serializer for register new user
+    """
+
+    username = serializers.CharField(max_length=100, required=True)
+    password = serializers.CharField(max_length=500, required=True)
```
Comment (artsiom-streetshares) [inline app/api/auth/serializers.py]: why that much?

File: app/api/auth/serializers.py [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+    def validate_email(email: str) -> str:
+        return validate_email_field(email=email)
+
+    @staticmethod
+    def validate_password(password: str) -> str:
+        return validate_password_field(password=password)
+
+    def validate(self, attrs: tp.Dict[str, tp.Any]) -> tp.Dict[str, tp.Any]:
+        """
+        Validate all fields
```
Comment (artsiom-streetshares) [inline app/api/auth/serializers.py]: it doesn't validate all fields

File: app/api/auth/views.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+class RegisterUserApiView(APIView):
+    """
+    Api View for creating new user
+    """
+
+    @swagger_auto_schema(request_body=RegisterUserSerializer)
+    def post(self, request: Request, *args: tp.Any, **kwargs: tp.Any) -> Response:
+        serializer = RegisterUserSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        serializer_data = serializer.data
```
Comment (artsiom-streetshares) [inline app/api/auth/views.py]: I would either create a new layer around serializers and call it from here or included in into services

Comment (artsiom-streetshares) [reply app/api/auth/views.py]: like the layer would return serialized and validated data and services layer would take it as input

File: app/api/auth/validation.py [hunk 6]
```diff
@@ -0,0 +1,55 @@
+from rest_framework import serializers
```
Comment (artsiom-streetshares) [inline app/api/auth/validation.py:1]: I think its not really useful to separate out these functions from serializer, unless you want to reuse them.

File: app/api/video/migrations/0001_initial.py [hunk 7]
```diff
@@ -15,10 +15,11 @@ class Migration(migrations.Migration):
             name='Video',
             fields=[
                 ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
-                ('title', models.CharField(max_length=100)),
-                ('description', models.TextField(max_length=1024)),
-                ('bucket_path', models.CharField(max_length=100)),
-                ('upload_date', models.DateTimeField(auto_now_add=True)),
+                ('title', models.CharField(max_length=100, verbose_name='title')),
```
Comment (artsiom-streetshares) [inline app/api/video/migrations/0001_initial.py:18]: you should never ever change existing migrations in your PR, just add a new one. research why changing migrations is bad

File: .gitignore (hunk 2 above)
Comment (xal9wiii4ik) [reply .gitignore]: sorry, i found in google gitignore file for poetry

Comment (xal9wiii4ik) [reply .gitignore]: done

File: app/api/auth/serializers.py (hunk 3 above)
Comment (xal9wiii4ik) [reply app/api/auth/serializers.py]: yes, max length for password is really too much, i forgot this moment

File: app/api/auth/validation.py (hunk 6 above)
Comment (xal9wiii4ik) [reply app/api/auth/validation.py:1]: yes, it is reused in UpdateAccountSerializer

File: app/api/auth/views.py (hunk 5 above)
Comment (xal9wiii4ik) [reply app/api/auth/views.py]: done

File: app/api/auth/serializers.py (hunk 4 above)
Comment (xal9wiii4ik) [reply app/api/auth/serializers.py]: i have update func description but i dont think that new description looks good

File: app/api/video/migrations/0001_initial.py (hunk 7 above)
Comment (xal9wiii4ik) [reply app/api/video/migrations/0001_initial.py:18]: ok, I got it, apparently I misunderstood `squash`

File: app/api/auth/serializers.py (hunk 3 above)
Comment (xal9wiii4ik) [reply app/api/auth/serializers.py]: done

**Pull request closed without merging**